src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -366,53 +228,10 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
             }
 
             var result = razorRanges.ToImmutableList();
-            return new VersionedSemanticRange(result, csharpResponse.IsFinalizedCSharp);
-        }
-
-        // Internal for testing only
-        internal static bool TryGetMinimalCSharpRange(RazorCodeDocument codeDocument, [NotNullWhen(true)] out Range? range)
-        {
-            var csharpDoc = codeDocument.GetCSharpDocument();
-
-            // If there aren't any source mappings, there's no C# code in the Razor doc.
-            if (csharpDoc.SourceMappings.Count == 0)
-            {
-                range = null;
-                return false;
-            }
-
-            // We only need to colorize the portions of the generated doc that correspond with a C# mapping.
-            // To accomplish this while minimizing the amount of work we need to do, we'll only colorize the
-            // range spanning from the first C# span to the last C# span.
-            SourceSpan? minSpan = null;
-            SourceSpan? maxSpan = null;
-
-            foreach (var mapping in csharpDoc.SourceMappings)
-            {
-                var generatedSpan = mapping.GeneratedSpan;
-                if (minSpan is null || generatedSpan.AbsoluteIndex < minSpan.Value.AbsoluteIndex)
-                {
-                    minSpan = generatedSpan;
-                }
-
-                if (maxSpan is null || generatedSpan.AbsoluteIndex + generatedSpan.Length > maxSpan.Value.AbsoluteIndex + maxSpan.Value.Length)
-                {
-                    maxSpan = generatedSpan;
-                }
-            }
-
-            var csharpSourceText = codeDocument.GetCSharpSourceText();
-            var start = csharpSourceText.Lines.GetLinePosition(minSpan!.Value.AbsoluteIndex);
-            var startPosition = new Position(start.Line, start.Character);
-
-            var end = csharpSourceText.Lines.GetLinePosition(maxSpan!.Value.AbsoluteIndex + maxSpan!.Value.Length);
-            var endPosition = new Position(end.Line, end.Character);
-
-            range = new Range(startPosition, endPosition);
-            return true;
+            return new List<SemanticRange>(result);
```

Comment:
`razorRanges` is a List that gets converted to an ImmutableList that then gets wrapped back into a List.. just return it?